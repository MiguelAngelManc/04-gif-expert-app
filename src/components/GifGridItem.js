import React from 'react'
import PropTypes from 'prop-types'

const GifGridItem = ({id, title, url}) => {
    
    
    return (
        <div className="card animate__animated animate__fadeIn">
            <p>{title}</p>
            <img src={url} />
        </div>
    )
}

GifGridItem.propTypes = {

}

export default GifGridItem
