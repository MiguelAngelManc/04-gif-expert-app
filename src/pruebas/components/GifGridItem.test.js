import { React } from "react";
import { render } from "@testing-library/react";
import GifGridItem from "../../components/GifGridItem";
import { shallow } from "enzyme";

describe('Pruebas de GifGridItem',()=>{

    const title = 'Un titulo'
    const url ="https://UnaLiga.com/archivo.jpg"

    const wrapper = shallow(<GifGridItem title={title} url={url}/>)


    test('Prueba de snapshot',()=>{
        expect(wrapper).toMatchSnapshot();
    })

    test('Leyendo el titulo',()=>{
        const p = wrapper.find('p').text();
        expect(p.trim()).toBe(title)
    })

    test('Leyendo la imagen',()=>{
        const src = wrapper.find('img').prop('src');
        const alt = wrapper.find('img').prop('alt');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    })

    test('Debe tener la clase de card',()=>{
        const div = wrapper.find('div')

        expect(div.hasClass('card')).toBe(true);

    })

})