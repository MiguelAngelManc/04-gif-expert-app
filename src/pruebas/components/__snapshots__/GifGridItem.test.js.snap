// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GifGridItem Prueba de snapshot 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <p>
    Un titulo
  </p>
  <img
    alt="Un titulo"
    src="https://UnaLiga.com/archivo.jpg"
  />
</div>
`;
